### PR TITLE
Icu 13218 update indexed db parser unit tests to test for filtering on boolean fields

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-13218)

## Description
Silence deprecation warnings and add test cases for bool fields.

before:

image

after:

image

## How to Test

Run unit tests and make sure no deprecation warnings are shown in dev tools console

## Checklist:

- [X] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
LGTM

File: addons/api/tests/unit/utils/indexed-db-query-test.js [hunk 1]
```diff
@@ -494,4 +496,32 @@ module('Unit | Utility | indexed-db-query', function (hooks) {
...
+
+    assert.strictEqual(result.length, 1);
+  });
+
+  test('it filters on boolean type fields with falsy value', async function (assert) {
+    const authMethod = this.server.create('auth-method', {
+      name: 'Generated_auth_method',
+      scopeId: scope1.id,
+    });
+    scope1.update({ primaryAuthMethodId: authMethod.id });
```
Comment (ZedLi) [inline addons/api/tests/unit/utils/indexed-db-query-test.js:519]: Does this actually set the `is_primary` for the auth method itself? I don't think we have anything in the config that would update the auth method model when we patch this

**Review by ZedLi — APPROVED** [review]
Thanks for adding more tests!

File: addons/api/tests/unit/utils/indexed-db-query-test.js (hunk 1 above)
Comment (lisbet-alvarez) [reply addons/api/tests/unit/utils/indexed-db-query-test.js:519]: yeah, not in the config, only in the mirage [auth-method serializer ](https://github.com/hashicorp/boundary-ui/blob/main/addons/api/mirage/serializers/auth-method.js#L20)

**Pull request merged**